tests/backup/locations_test.js

```diff
@@ -0,0 +1,369 @@
+
+    // Verify S3 is selected
+    I.seeAttributesOnElements(
+      locationsPage.buttons.typeSelect(locationsPage.locationType.s3),
+      { checked: true },
+    );
+    I.seeAttributesOnElements(
+      locationsPage.buttons.typeSelect(locationsPage.locationType.client),
+      { checked: null },
+    );
+    I.seeAttributesOnElements(
+      locationsPage.buttons.typeSelect(locationsPage.locationType.server),
+      { checked: null },
+    );
+
+    // Verify buttons state
+    I.seeTextEquals('Add', locationsPage.buttons.addLocation);
+    I.seeAttributesOnElements(locationsPage.buttons.addLocation, { disabled: true });
+    I.seeTextEquals('Test', locationsPage.buttons.testLocation);
+    I.seeAttributesOnElements(locationsPage.buttons.testLocation, { disabled: true });
+    I.seeTextEquals('Cancel', locationsPage.buttons.cancel);
+    I.seeAttributesOnElements(locationsPage.buttons.cancel, { disabled: null });
+  },
+);
+
+Scenario(
+  'PMM-T696 Verify validation for add storage location modal @backup',
+  async ({
+    I, locationsPage,
+  }) => {
+    locationsPage.openAddLocationModal();
+
+    Object.entries(locationsPage.fields)
+      .filter(([key]) => key !== 'description')
+      .forEach(([, locator]) => {
+        I.click(locator);
+      });
+
+    I.click(locationsPage.fields.description);
+
+    Object.entries(locationsPage.elements.validation)
+      .forEach(([, locator]) => {
+        I.seeTextEquals(locationsPage.messages.requiredField, locator);
+      });
+  },
+);
+
+Scenario(
+  'PMM-T707 Verify user is able to test connection for storage location @backup',
+  async ({
+    I, locationsPage,
+  }) => {
+    const location = {
+      name: 'Location',
+      description: 'test description',
+      ...locationsPage.storageLocationConnection,
+    };
+
+    locationsPage.openAddLocationModal();
+
+    // Fill required fields
+    locationsPage.fillLocationFields(location);
+
+    // Verify buttons become enabled
+    I.seeAttributesOnElements(locationsPage.buttons.addLocation, { disabled: null });
+    I.seeAttributesOnElements(locationsPage.buttons.testLocation, { disabled: null });
+
+    // Verify Test Connection works
+    I.click(locationsPage.buttons.testLocation);
+    I.verifyPopUpMessage(locationsPage.messages.successfullyTested);
+  },
+);
+
+Data(s3Errors).Scenario(
+  'PMM-T708 Verify errors related to s3 storage location @backup',
+  async ({
+    I, locationsPage, current,
+  }) => {
+    const location = {
+      name: 'Location',
+      ...locationsPage.storageLocationConnection,
+    };
+
+    locationsPage.openAddLocationModal();
+    locationsPage.fillLocationFields({ ...location, [current.field]: current.value });
+    I.click(locationsPage.buttons.testLocation);
+    I.verifyPopUpMessage(current.error, 40);
+  },
+);
+
+Scenario(
+  'PMM-T683 PMM-T684 Verify user is able to add storage location @backup',
+  async ({
+    I, locationsPage,
+  }) => {
+    const location = {
+      name: 'Location',
```

Comment:
could you think about randomize the location name? in case cleanup fails, it will avoid false positives due to it

Reply:
done, thanks